lib/ace/edit_session/bracket_match.js

```diff
@@ -37,10 +37,77 @@ var Range = require("../range").Range;
 
 function BracketMatch() {
 
+
+    this.getBracketsCoor = function(position) {
+        if (position.column == 0) return null;
+
+        var line = this.getLine(position.row);
+
+
+        var charBeforeCursor,
+            matchLeft,
+            charAfterCursor,
+            matchRight;
+        
+        var length = -1,
+            depthLeft = 0,
+            depthRight = 0;
+
+        var regExpOpenBrackets = /([\(\[\{])/,
+            regExpCloseBrackets = /([\)\]\}])/;
+
+
+        while (!matchLeft && !matchRight) {
+
+            length++;
+
+            charBeforeCursor = line.charAt(position.column - length);
+            charAfterCursor = line.charAt(position.column + length);
+
+
+            if (charBeforeCursor.match(regExpCloseBrackets)) {
+                depthLeft++;
+            } else { 
+                matchLeft = charBeforeCursor.match(regExpOpenBrackets);
+                if ((matchLeft) && (depthLeft !== 0)) {
+                    matchLeft = null;
+                    depthLeft--;
+                }
+            }
+
+           
+
+            if (charAfterCursor.match(regExpOpenBrackets)) {
+                depthRight++;
+            } else {
+                matchRight = charAfterCursor.match(regExpCloseBrackets);
+                if ((matchRight) && (depthRight !== 0)) {
+                    matchRight = null;
+                    depthRight--;
+                }   
+            }
+
+            if (!charBeforeCursor || !charAfterCursor) 
+                return null;
+                
+        }
+
+        if (matchLeft) {
+            position.column -= length - 1;
```

Comment:
this line where you decrease position.column by length-1 and subsequent line where you subtract -1 once again look weird. Can you just return `column: position.column - length` ?